src/mcu_stm8s.c

```diff
@@ -285,6 +285,9 @@ static void UART_setup(void)
     UART1_PARITY_NO,
     UART1_SYNCMODE_CLOCK_DISABLE,
     UART1_MODE_TXRX_ENABLE);
+		
+  UART1_ITConfig(UART1_IT_RXNE_OR, ENABLE); 
+  enableInterrupts();	
```

Comment:
Interrupts are to be enabled only after all peripherals are initialized - enableInterrupts() to be called from main only after all the inits are done, otherwise chaos may ensue!